### ISSUE TITLE
SQlite database lock

### ISSUE THREAD

**Issue by Niohuruzh** [nextgenusfs/funannotate#745]
Hi funannotate team
 When I use `funannotate train` , I get trouble whether on docker or local as follow command: 
`funannotate train -i W3767_clean_sort_mask.fasta -o fun --left 105-1/105-1_1.fq.gz --right 105-1/105-1_2.fq.gz --stranded RF --jaccard_clip --species cryptococcus_neoformans_gattii --cpus 4 --strain Cgattii --isolate 105`

```
-in SQLite mode, disabling multithreading
DBD::SQLite::db prepare failed: database is locked at /home/anaconda3/envs/funannotate/opt/pasa-2.5.2/PerlLib/DB_connect.pm line 160.
Error DBD::SQLite::db prepare failed: database is locked at /home/anaconda3/envs/funannotate/opt/pasa-2.5.2/PerlLib/DB_connect.pm line 160.
 at /home/anaconda3/envs/funannotate/opt/pasa-2.5.2/PerlLib/DB_connect.pm line 162.
	DB_connect::do_sql_2D(DB_connect=HASH(0x55f1012c5cb8), "select al.align_id, al.align_acc from subcluster_link sl, ali"..., 8) called at /home/leigod/anaconda3/envs/funannotate/opt/pasa-2.5.2/scripts/classify_alt_splice_isoforms_per_subcluster.dbi line 63
Thread 6 terminated abnormally: Error, command /home/anaconda3/envs/funannotate/opt/pasa-2.5.2/scripts/classify_alt_splice_isoforms_per_subcluster.dbi -M /media/f83cb121-4d63-4d87-8adf-a7bdefb24a63/home/data/szh/gattii_genome/fun/training/pasa/cryptococcus_neoformans_gattii_Cgattii_pasa -s 8 > __tmp_classify_alt_isoforms/8.out died with ret (2816). Maybe see __tmp_classify_alt_isoforms/8.out.temp for information at /home/anaconda3/envs/funannotate/opt/pasa-2.5.2/scripts/classify_alt_splice_isoforms.dbi line 290.
ERROR, thread 6 exited with error Error, command /home/anaconda3/envs/funannotate/opt/pasa-2.5.2/scripts/classify_alt_splice_isoforms_per_subcluster.dbi -M /media/f83cb121-4d63-4d87-8adf-a7bdefb24a63/home/data/szh/gattii_genome/fun/training/pasa/cryptococcus_neoformans_gattii_Cgattii_pasa -s 8 > __tmp_classify_alt_isoforms/8.out died with ret (2816). Maybe see __tmp_classify_alt_isoforms/8.out.temp for information at /home/anaconda3/envs/funannotate/opt/pasa-2.5.2/scripts/classify_alt_splice_isoforms.dbi line 290.
```
So, if I set cpus=1, trinity is not work. If I set cpus=2, Sqlite database will lock. Is there a sufficient way to solve it? BTW my funannotate version is 1.8.11

LOOKING FORWARD YOUR REPLY

**Comment by nextgenusfs** [comment]
Hmm. I think this could be an issue with PASA version you are using. I thought this was fixed in v2.4.1 - where it would only use single thread if using SQLite. You can try an older version of PASA or if you are able to use MySQL as backend to pass than that will be faster as it is multithreaded.

**Comment by Niohuruzh** [comment]
Thanks for your reply.
I used PASA v2.5.2 and encountered this problem. And now I try PASA v2.4.1. I hope it will work will, or I will download mysql to use PASA.
